### PR TITLE
fast endpoint + preferences

### PR DESCRIPTION
**Pull request by bhakiyakalimuthu**
- add preference to sendprivate tx request
- update tests

### CODE REVIEW

**Review by Ruteri — APPROVED** [review]

File: server/request_handler.go [hunk 1]
```diff
@@ -59,6 +60,14 @@ func (r *RpcRequestHandler) process() {
 		return
 	}
 
+	var isFast bool
```
Comment (Ruteri) [inline server/request_handler.go:63]: Nit: I'm guessing this can be declared inside if block

Comment (bhakiyakalimuthu) [reply server/request_handler.go]: Done.

File: types/types.go [hunk 2]
```diff
@@ -103,3 +104,13 @@ type BundleResponse struct {
 	BundleId string   `json:"bundleId"`
 	RawTxs   []string `json:"rawTxs"`
 }
+
+type SendPrivateTransactionRequestWithPreference struct {
+	SendPrivateTx flashbotsrpc.FlashbotsSendPrivateTransactionRequest
```
Comment (metachris) [inline types/types.go:109]:
```suggestion
	flashbotsrpc.FlashbotsSendPrivateTransactionRequest
```

File: types/types.go [hunk 3]
```diff
@@ -103,3 +104,13 @@ type BundleResponse struct {
 	BundleId string   `json:"bundleId"`
 	RawTxs   []string `json:"rawTxs"`
 }
+
+type SendPrivateTransactionRequestWithPreference struct {
```
Comment (metachris) [inline types/types.go:108]:
```suggestion
type SendPrivateTransactionRequestWithPreferences struct {
```

File: types/types.go [hunk 4]
```diff
@@ -103,3 +104,13 @@ type BundleResponse struct {
...
+	SendPrivateTx flashbotsrpc.FlashbotsSendPrivateTransactionRequest
+	Preferences   *Preferences `json:"preference,omitempty"`
+}
+
+type Preferences struct {
```
Comment (metachris) [inline types/types.go:113]:
```suggestion
type PrivateTxPreferences struct {
```

File: testutils/mock_rpcbackend.go [hunk 5]
```diff
@@ -67,7 +68,11 @@ func handleRpcRequest(req *types.JsonRpcRequest) (result interface{}, err error)
 
 		// Relay calls
 	case "eth_sendPrivateTransaction":
-		param := req.Params[0].(map[string]interface{})
+		reqParam := req.Params[0].(map[string]interface{})
+		param := reqParam["SendPrivateTx"].(map[string]interface{})
+		if preference, ok := reqParam["preference"]; ok {
```
Comment (metachris) [inline testutils/mock_rpcbackend.go:73]: use plural

File: server/request_handler.go (hunk 1 above)
Comment (metachris) [reply server/request_handler.go]: it's not needed at all

File: server/request_handler.go [hunk 6]
```diff
@@ -59,6 +60,13 @@ func (r *RpcRequestHandler) process() {
 		return
 	}
 
+	var preferences *types.Preferences
```
Comment (metachris) [inline server/request_handler.go:63]: wouldn't even need to be a pointer, could instantiate here already and have every private tx have preferences. wdyt? 🤔

File: server/request_handler.go [hunk 7]
```diff
@@ -59,6 +60,13 @@ func (r *RpcRequestHandler) process() {
...
+	if strings.Trim(r.req.URL.Path, "/") == "fast" { // If fast called, do not include tx to bundle, directly send tx to miners
```
[the remark this answers is not in the document]
Comment (bhakiyakalimuthu) [reply server/request_handler.go:64]: I want to make sure if the user uses double slashes `baseUrl/fast/`, we should be able to handle

File: server/request_handler.go [hunk 8]
```diff
@@ -59,6 +60,13 @@ func (r *RpcRequestHandler) process() {
...
+		isFast := true
+		preferences = &types.Preferences{Fast: &isFast}
```
[the remark this answers is not in the document]
Comment (bhakiyakalimuthu) [reply server/request_handler.go:66]: I want to keep it as pointer as going forward we will have multiple preferences.

File: server/request_handler.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (metachris) [reply server/request_handler.go:64]: makes sense!

File: types/types.go [hunk 9]
```diff
@@ -103,3 +104,13 @@ type BundleResponse struct {
...
+	// If set do not include tx to bundle, directly send tx to miners
+	Fast *bool `json:"fast,omitempty"`
```
[the remark this answers is not in the document]
Comment (bhakiyakalimuthu) [reply types/types.go:115]: I want to keep it as pointer as going forward we will have multiple preferences. When we have auction included we want to omit fast.What do you think ?

File: server/request_handler.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (metachris) [reply server/request_handler.go:66]: why does a pointer help? seems to me like an argument for not being a pointer and already instantiated 🤔

File: types/types.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (metachris) [reply types/types.go:115]: I think it should be either true or false and always sent. What's an advantage of having this be a pointer?

[the remark this answers is not in the document]
Comment (bhakiyakalimuthu) [reply types/types.go:115]: ok.makes sense

File: server/request_handler.go (hunk 1 above)
Comment (bhakiyakalimuthu) [reply server/request_handler.go]: updated

**Review by metachris — APPROVED** [review]
lgtm

**Pull request merged**